gslib/command_runner.py

```diff
@@ -410,7 +411,14 @@ def RunNamedCommand(self,
                                sub_opts=command_inst.sub_opts,
                                command_alias=command_name)
 
-    return_code = command_inst.RunCommand()
+    if command_inst.translate_to_gcloud_storage_if_requested():
```

Comment:
The method is on `Command`, so you can read it as "translate the command to gcloud storage". Explaining what mechanism decides if the translation will happen in the function name will make it too long which is already pretty lengthy, and it is hard to capture all the mechanisms anyway. So I think the docstring is apt for that use case.

Reply:
It seems like the naming is reflecting an underlying code smell.
This structure might be better:
```
if command_inst.should_translate_to_gcloud_storage():
  command_inst.translate_to_gcloud_storage()
  ...
```

Reply:
Yeah, I had thought of breaking it down that way, but I did not do it because the conditions on whether to call gsutil or gcloud are based on whether the translation was happened successfully as well, along with knowing whether we should perform the translation. 

Keeping it this way help us do the following in the command runner.
```
    if command_inst.translate_to_gcloud_storage_if_requested():
      # This does not mean that the gcloud storage command worked.
      # It only means that we succesfully attempted running gcloud storage.
      # The command itself might have failed.
      return_code = command_inst.run_gcloud_storage()
    else:
      # Run gsutil.
      return_code = command_inst.RunCommand()
```

I had used the structure that you've mentioned earlier and it led to too much handling in command_runner (even with using helper) and I wanted to keep the command_runner code as clean as possible. Ideally, I wanted it to do "if condition 'call gcloud storage' else 'call gsutil' "

This is just a cost of keeping the command_runner straight-forward, which I feel is not a bad trade-off

Reply:
That's fine. I guess it avoids:
```
return_code = None
if is_configured_to_translate():
  did_translation = attempt_translation()
  if did_translation:
    return_code = run_translation()
if not return_code:
  return_code = run_original()
```